src/github.com/getlantern/lantern-android/client/client_test.go

```diff
@@ -0,0 +1,108 @@
+package client
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"testing"
+	"time"
+)
+
+const listenProxyAddr = "127.0.0.1:9997"
+
+var testURLs = map[string][]byte{
+	"http://www.google.com/humans.txt": []byte("Google is built by a large team of engineers, designers, researchers, robots, and others in many different sites across the globe. It is updated continuously, and built with more tools and technologies than we can shake a stick at. If you'd like to help us out, see google.com/careers.\n"),
+	// TODO: This is not working, we actually need to implement a CONNECT request to proxy HTTPs traffic.
+	// "https://www.google.com/humans.txt": []byte("Google is built by a large team of engineers, designers, researchers, robots, and others in many different sites across the globe. It is updated continuously, and built with more tools and technologies than we can shake a stick at. If you'd like to help us out, see google.com/careers.\n"),
```

Comment:
@xiam You should be able to configure the http.Transport with a proxy, and it will take care of doing the CONNECT request for you.